clean_helpers/deduplication.py

```diff
@@ -72,7 +74,8 @@ def dedup_document(ds: Dataset, num_proc: int, batch_size: int) -> Dataset:
 
 def get_hash(texts: List[str]) -> List[str]:
     """Get hash of content field."""
-    return [hashlib.md5(text.strip().encode("utf-8")).hexdigest() for text in texts]
+    stripped_texts = [re.sub(f'\s+|\d+|[{re.escape(string.punctuation)}]','', text) for text in texts]
```

Comment:
This will screw with math btw. But then I don't think it would actually anything in natural language?
I'm not sure how many more hashes we actually match ... If you had an example I could understand better?

Reply:
Also since we're going to run this quite a bit. Let's compile it
```
remove_non_character_regex = re.compile(f'\s+|\d+|[{re.escape(string.punctuation)}]')
def get_hash():
    stripped_texts = [remove_non_character_regex.sub('', text) for text in texts]
```